### DISCUSSION TITLE
Web-component to component user communication

### DISCUSSION THREAD

**Discussion by vdwivedi-mdsol** [bitovi/react-to-web-component#150] [Q&A] [answered]
I have created the web component using react-to-web-component library lets say `<my-webcomponent>,` and using it  in a application , lets say that application is  **HOST_APP** 

how `<my-webcomponent>`will inform **HOST_APP** that it has done his job .

**Comment by christopherjbaker** [comment, marked as answer]
Hey. Great question. I am working on a guide and a library for exactly this! The library will make it easy to provide event listeners that tie into the component following familiar React-like patterns.

You can use use a state management system like Redux to coordinate your whole application (guide on this also coming soon), but it sounds like that would be overkill for this project.

In the meantime, the most straightforward way is to define a global function (attached to the window) and tell r2wc the name of that function. Then you can call it within your component as you would any other function prop.

### React Component
```tsx
const MyComponent: React.FC<{ onComplete: () => void }> = ({ onComplete }) => {
  return <button onClick={onComplete}>Click me!</button>
})
```

### R2WC Definition
```jsx
customElements.define(
  "my-webcomponent",
  r2wc(MyComponent, {
    props: {
      onComplete: "function"
    },
  }),
)
```

### Usage
```js
window.myFunction = () => {
  console.log("Done!")
}
```

```html
<my-webcomponent on-complete="myFunction"></my-webcomponent>
```
